### PR TITLE
remove bower

### PR DESCRIPTION
**Pull request by GnsP**
# Remove bower

## Description

TLDR;
Move the dependencies managed by bower to yarn and remove bower.

Currently, some of the dependencies (legacy) are being managed by bower. However, bower is not in active development and it's recommended to migrate to yarn. In this PR we have migrated the dependencies and removed bower, with the following changes to the codebase:

1. Equivalent versions of the dependencies managed by bower have been added to `package.json` to be managed by `yarn >= 1.0`. 
2. `bower_components` have been moved to `node_modules/@bower_components`. All references to `bower_components` in the codebase have been updated accordingly.
3. The script `bower_root` has been deprecated and the corresponding step in the github build workflow has been removed.
4. `README.md` has been updated to remove the `bower install` step.

## PR Type
- [ ] Bug Fix
- [ ] Feature
- [x] Build Fix
- [ ] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [CDAP-14285](https://cdap.atlassian.net/browse/CDAP-14285)

## Test Plan

Local build: pass
Github build workflow: pass
e2e tests: pass
Manual tests: wip

## Screenshots

NA

[CDAP-14285]: https://cdap.atlassian.net/browse/CDAP-14285?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by seanfeld — APPROVED** [review]

File: Gulpfile.js [hunk 1]
```diff
@@ -100,13 +100,12 @@ function getBabelBuildPipeline() {
...
   return merge(
     gulp.src([
-      './bower_components/angular/angular-csp.css',
-      './bower_components/angular-loading-bar/build/loading-bar.min.css',
-      './bower_components/angular-motion/dist/angular-motion.min.css',
+      './node_modules/@bower_components/angular/angular-csp.css',
+      './node_modules/@bower_components/angular-loading-bar/build/loading-bar.min.css',
+      './node_modules/@bower_components/angular-motion/dist/angular-motion.min.css',
       './node_modules/font-awesome/css/font-awesome.min.css',
-      './bower_components/c3/c3.min.css',
```
Comment (radhikav1) [inline Gulpfile.js:107]: seems c3.min.css is removed, is this intentional? If so, are we sure that it is not used anywhere in the code?

File: Gulpfile.js [hunk 2]
```diff
@@ -158,74 +157,73 @@ gulp.task('css:app', gulp.series('css:application', 'css:lint'));
...
+
+      './node_modules/@bower_components/angular-cookies/angular-cookies.min.js',
       './node_modules/redux/dist/redux.min.js',
       './node_modules/redux-thunk/dist/redux-thunk.min.js',
-      './bower_components/ace-builds/src-min-noconflict/ace.js',
-      './bower_components/angular-ui-ace/ui-ace.js',
-      './bower_components/jsPlumb/dist/js/jsPlumb-2.0.6-min.js',
-      './bower_components/angular-gridster/dist/angular-gridster.min.js',
-      './bower_components/angular-cron-jobs/dist/angular-cron-jobs.min.js',
-      './bower_components/angularjs-dropdown-multiselect/dist/angularjs-dropdown-multiselect.min.js',
```
Comment (radhikav1) [inline Gulpfile.js:220]: is this also removed?

**Review by njbriggs — APPROVED** [review]

**Review by radhikav1 — APPROVED** [review]

File: Gulpfile.js (hunk 1 above)
Comment (GnsP) [reply Gulpfile.js:107]: yes, we have checked the codebase and found that the `c3-` css classes are not being used anywhere.

File: Gulpfile.js (hunk 2 above)
Comment (GnsP) [reply Gulpfile.js:220]: yes, this component is not in use anymore. It has been replaced by a newer `multiselect` component written in react.
